03_Modules/Transactions/src/module/module.ts

```diff
@@ -113,98 +109,7 @@ export class TransactionsModule extends BaseModule {
-                new Date() > new Date(idTokenInfo.cacheExpiryDateTime)) {
-                response.idTokenInfo = {
-                  status: AuthorizationStatusEnumType.Invalid,
-                  groupIdToken: idTokenInfo.groupIdToken
-                  // TODO determine how/if to set personalMessage
-                };
-              } else {
-                // TODO: Determine how to check for NotAllowedTypeEVSE, NotAtThisLocation, NotAtThisTime, NoCredit
-                // TODO: allow for a 'real time auth' type call to fetch token status.
-                response.idTokenInfo = idTokenInfo;
-              }
-            } else {
-              // IdTokenInfo.status is one of Blocked, Expired, Invalid, NoCredit
-              // N.B. Other non-Accepted statuses should not be allowed to be stored.
-              response.idTokenInfo = idTokenInfo;
-            }
-          } else {
-            // Assumed to always be valid without IdTokenInfo
-            response.idTokenInfo = {
-              status: AuthorizationStatusEnumType.Accepted
-              // TODO determine how/if to set personalMessage
-            };
-          }
-        }
-        return response;
-      }).then(transactionEventResponse => {
-        if (transactionEvent.eventType == TransactionEventEnumType.Started && transactionEventResponse
-          && transactionEventResponse.idTokenInfo?.status == AuthorizationStatusEnumType.Accepted && transactionEvent.idToken) {
-          // Check for ConcurrentTx
-          return this._transactionEventRepository.readAllActiveTransactionByIdToken(transactionEvent.idToken).then(activeTransactions => {
-            // Transaction in this TransactionEventRequest has already been saved, so there should only be 1 active transaction for idToken
-            if (activeTransactions.length > 1) {
-              const groupIdToken = transactionEventResponse.idTokenInfo?.groupIdToken;
-              transactionEventResponse.idTokenInfo = {
-                status: AuthorizationStatusEnumType.ConcurrentTx,
-                groupIdToken: groupIdToken
-                // TODO determine how/if to set personalMessage
-              }
-            }
-            return transactionEventResponse;
-          });
-        }
-        return transactionEventResponse;
-      }).then(transactionEventResponse => {
-        this.sendCallResultWithMessage(message, transactionEventResponse)
-          .then(messageConfirmation => this._logger.debug("Transaction response sent: ", messageConfirmation));
-      });
-    } else {
-      const response: TransactionEventResponse = {
-        // TODO determine how to set chargingPriority and updatedPersonalMessage for anonymous users
-      };
-      this.sendCallResultWithMessage(message, response)
-        .then(messageConfirmation => this._logger.debug("Transaction response sent: ", messageConfirmation));
-    }
+    await this.transactionEventService?.handleTransactionEvent(message, props);
```

Comment:
This is the main drive behind the changes. We have a module with multiple handlers, and here we can have a way to have multiple services handling the events which is greatly simplified with DI since things that are needed by the class can be imported into the constructor automatically